modules/4337/test/erc4337/ERC4337ModuleNew.spec.ts

```diff
@@ -115,16 +121,33 @@ describe('Safe4337Module - Newly deployed safe', () => {
         safeOp,
         signature,
       })
-      await entryPoint.executeUserOp(userOp, 0)
-      expect(await ethers.provider.getBalance(safe.address)).to.be.eq(ethers.parseEther('0.5'))
-      await expect(entryPoint.executeUserOp(userOp, 0)).to.be.revertedWithCustomError(entryPoint, 'InvalidNonce').withArgs(0)
+
+      const newSafeOp = buildSafeUserOpTransaction(
+        safe.address,
+        user1.address,
+        ethers.parseEther('0.5'),
+        '0x',
+        '0',
+        await entryPoint.getAddress(),
+        false,
+        false,
+      )
+      await entryPoint.handleOps([userOp], user1.address)
+      const newSignature = buildSignatureBytes([await signSafeOp(user1, await validator.getAddress(), newSafeOp, await chainId())])
+      const newUserOp = buildUserOperationFromSafeUserOperation({
+        safeOp: newSafeOp,
+        signature: newSignature,
+      })
+
+      await expect(entryPoint.handleOps([newUserOp], user1.address))
+        .to.be.revertedWithCustomError(entryPoint, 'FailedOp')
+        .withArgs(0, 'AA25 invalid account nonce')
```

Comment:
Why is the `newUserOp` needed instead of just trying to replay `userOp`?

Reply:
Because the other user operation contains `initCode` and it will revert with an error "sender already constructed" and I don't think this is what we're trying to validate here

Reply:
:bulb:

Reply:
On second thought, do you think it makes sense to `await safe.deploy()` so that there is no init code and really make this test "you can't replay a user operation" (which I feel was the original intent of the test)?

Not blocking PR, and I'll leave it up to you whether or not to incorporate this suggestion.

Reply:
I didn't know about this method, implemented it here and optimized a few other tests